### PR TITLE
 Set flag color to gray when not-top-site is the only signal

### PR DESCRIPTION
**Pull request by livvielin**

### CODE REVIEW

File: extension/background.js [hunk 1]
```diff
@@ -106,28 +106,40 @@ class Background {
...
-          '48': 'images/orangeflag48.png',
-          '128': 'images/orangeflag128.png',
+          '16': `images/${color}flag16.png`,
+          '48': `images/${color}flag48.png`,
+          '128': `images/${color}flag128.png`,
         },
         tabId: tab.id,
       });
+    };
+
```
Comment (estark37) [inline extension/background.js]: optional nit: I might rewrite the below code with an early-return to reduce nesting:

```
if (alertList.length == 0) {
  setFlagIconColor('green', tab);
  return;
}
if (alertList.length == 1 && alertList.includes(...)) {
  setFlagIconColor('gray', tab);
} else {
...
  setFlagIconColor('orange', tab);
}
```

(this is just Chromium style so up to you if you want to follow it here!)

**Comment by estark37** [comment]
Oh, is this something that can be tested?

File: extension/background.js (hunk 1 above)
Comment (livvielin) [reply extension/background.js]: I like that styling--updated. Thanks!

**Comment by livvielin** [comment]
> Oh, is this something that can be tested?

Yes, it should be testable, but I think a bit tricky to get right. I imagine it'd have to be something along the lines of creating a spy for chrome.browserAction.setIcon/BadgeText and checking that it's called with the expected arguments--but IIRC doing this while some of the functions are nested in the background script didn't work out quite as expected. I think this might be best revisited after issue #16 is resolved.

**Comment by estark37** [comment]
> > Oh, is this something that can be tested?
> 
> Yes, it should be testable, but I think a bit tricky to get right. I imagine it'd have to be something along the lines of creating a spy for chrome.browserAction.setIcon/BadgeText and checking that it's called with the expected arguments--but IIRC doing this while some of the functions are nested in the background script didn't work out quite as expected. I think this might be best revisited after issue #16 is resolved.

Makes sense. Thanks!

**Pull request merged**